packaging/dbscripts/upgrade/pre_upgrade/0000_config.sql

```diff
@@ -1462,6 +1463,9 @@ select fn_db_update_default_config_value('ServerRebootTimeout', '300', '600', 'g
s":["200"]}}],"initialItems":[{"action":"setDowntime","params":["100"]}],"lastItems":[{"action":"postcopy","params":[]},{"action":"abort","params":[]}]}}]', '[{"id":{"uuid":"80554327-0569-496b-bdeb-fcbbf52b827b"},"maxMigrations":2,"autoConvergence":true,"migrationCompression":false,"enableGuestEvents":true,"name":"Minimal downtime","description":"","config":{"convergenceItems":[{"stallingLimit":1,"convergenceItem":{"action":"setDowntime","params":["150"]}},{"stallingLimit":2,"convergenceItem":{"action":"setDowntime","params":["200"]}},{"stallingLimit":3,"convergenceItem":{"action":"setDowntime","params":["300"]}},{"stallingLimit":4,"convergenceItem":{"action":"setDowntime","params":["400"]}},{"stallingLimit":6,"convergenceItem":{"action":"setDowntime","params":["500"]}}],"initialItems":[{"action":"setDowntime","params":["100"]}],"lastItems":[{"action":"abort","params":[]}]}},{"id":{"uuid":"80554327-0569-496b-bdeb-fcbbf52b827c"},"maxMigrations":1,"autoConvergence":true,"migrationCompression":true,"enableGuestEvents":true,"name":"Suspend workload if needed","description":"","config":{"convergenceItems":[{"stallingLimit":1,"convergenceItem":{"action":"setDowntime","params":["150"]}},{"stallingLimit":2,"convergenceItem":{"action":"setDowntime","params":["200"]}},{"stallingLimit":3,"convergenceItem":{"action":"setDowntime","params":["300"]}},{"stallingLimit":4,"convergenceItem":{"action":"setDowntime","params":["400"]}},{"stallingLimit":6,"convergenceItem":{"action":"setDowntime","params":["500"]}}],"initialItems":[{"action":"setDowntime","params":["100"]}],"lastItems":[{"action":"setDowntime","params":["5000"]},{"action":"abort","params":[]}]}},{"id":{"uuid":"a7aeedb2-8d66-4e51-bb22-32595027ce71"},"maxMigrations":2,"autoConvergence":true,"migrationCompression":false,"enableGuestEvents":true,"name":"Post-copy migration","description":"","config":{"convergenceItems":[{"stallingLimit":1,"convergenceItem":{"action":"setDowntime","params":["150"]}},{"stallingLimit":2,"convergenceItem":{"action":"setDowntime","params":["200"]}}],"initialItems":[{"action":"setDowntime","params":["100"]}],"lastItems":[{"action":"postcopy","params":[]},{"action":"abort","params":[]}]}},{"id":{"uuid":"57237b82-b8c2-425f-b425-114b35219626"},"maxMigrations":1,"autoConvergence":true,"migrationCompression":false,"enableGuestEvents":true,"zerocopy":true,"name":"Very large VMs","description":"","config":{"convergenceItems":[{"stallingLimit":1,"convergenceItem":{"action":"setDowntime","params":["400"]}},{"stallingLimit":2,"convergenceItem":{"action":"setDowntime","params":["500"]}},{"stallingLimit":3,"convergenceItem":{"action":"setDowntime","params":["700"]}},{"stallingLimit":4,"convergenceItem":{"action":"setDowntime","params":["1000"]}},{"stallingLimit":6,"convergenceItem":{"action":"setDowntime","params":["1500"]}}],"initialItems":[{"action":"setDowntime","params":["300"]}],"lastItems":[{"action":"setDowntime","params":["5000"]},{"action":"abort","params":[]}]}}]', '4.7', false);
```

Comment:
Why changing 4.7 MigrationPolicies only when they have the default value? Should thiis value be updated always? Or if this value can be customized by users, how users, who already created their own migration policies in 4.7, wil receive this new zerocopy policy?

Reply:
> Why changing 4.7 MigrationPolicies only when they have the default value? Should thiis value be updated always? 

It would overwrite contingent users' customizations.

> Or if this value can be customized by users, how users, who already created their own migration policies in 4.7, wil receive this new zerocopy policy?

I don't know.  Do we have a mechanism for that?  If not then I'd say that:

- Having the new policy is not important for all users.
- Users who change their policies take responsibility for incorporating updates from upstream.

Reply:
I agree with @mz-pdm.

Reply:
So this smells like bad design, there should probably be 2 options:
- `DefaultMigrationPolicies`, which would include our default migration policies and which wouldn't be modifiable by users
- `MigrationPolicies`, which would be modifiable by users
Having those 2 options and merging them together inside enginne backend would allow us easily update internal migration policies without affecting custom migration policies.

Unfortunately with current status we would need to add a description how to add zerocopy migration policy into 4.5.3 version if you already have your own custom policies

Reply:
Considering what's written in the comments above, I think splitting it to DefaultMigrationPolicies and MigrationPolicies wouldn't bring a real benefit.  If we were going to add more policies in the future, I'd be for reworking it some way (the option content is indeed horrible) but at the current state of matters it's not worth it.